src/cata_utility.h

```diff
@@ -10,6 +10,34 @@ class Creature;
 class map_item_stack;
 struct tripoint;
 
+/**
+* A class template that mimics a 2 dimensional array.
+*/
+template<typename T>
+class array_2d {
+private:
+    size_t size_x, size_y;
+    std::vector<T> _array;
+public:
+    array_2d( size_t x, size_t y ) : size_x( x ), size_y( y ), _array( std::vector<T>( x * y ) ) // Make sure capacity is sufficient
+    {
+    }
+
+    void set_at( size_t x, size_t y, T e )
+    {
+        if( x >= size_x || y >= size_y ) {
+            return;
+        }
+
+        _array[y * size_x + x] = e;
+    }
+
+    T get_at( size_t x, size_t y) const
+    {
+        return _array[y * size_x + x];
```

Comment:
This calculation of the the index should be extracted into a (possibly private) function and that function should be used in the setter and the getter. This makes sure both functions use the same logic for this.

Also: Why separate getters and setter? A function `at` (overloaded for const and non-const) that returns a reference would look better as it matches the standard vector / array classes.